augur/tasks/util/collection_util.py

```diff
@@ -36,6 +36,168 @@ class CollectionState(Enum):
+            AND secondary_data_last_collected <= NOW() - INTERVAL '{days_until_collect_again} DAYS'
+        """
+    
+    return condition_concat_string
+
+def get_required_conditions_for_facade_repos(allow_collected_before = False, days_until_collect_again = 1):
+
+    if not allow_collected_before:
+        condition_concat_string = f"""
+            facade_status='{str(CollectionState.UPDATE.value)}' AND facade_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.PENDING.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.FAILED_CLONE.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.INITIALIZING.value)}'
+            AND augur_operations.collection_status.facade_data_last_collected IS NULL
+            AND facade_status!='{str(CollectionState.COLLECTING.value)}'
+        """
+    else:
+        condition_concat_string = f"""
+            facade_status='Success' AND facade_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.facade_data_last_collected IS NOT NULL
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.PENDING.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.FAILED_CLONE.value)}'
+            AND augur_operations.collection_status.facade_status != '{str(CollectionState.INITIALIZING.value)}'
+            AND facade_status!='{str(CollectionState.COLLECTING.value)}'
+            AND facade_data_last_collected <= NOW() - INTERVAL '{days_until_collect_again} DAYS'
+        """
+    
+    return condition_concat_string
+
+def get_required_conditions_for_ml_repos(allow_collected_before = False, days_until_collect_again = 1):
+
+    if not allow_collected_before:
+        condition_concat_string = f"""
+            ml_status='{str(CollectionState.PENDING.value)}' AND ml_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.secondary_status = '{str(CollectionState.SUCCESS.value)}'
+            AND augur_operations.collection_status.ml_data_last_collected IS NULL
+            AND ml_status!='{str(CollectionState.COLLECTING.value)}'
+        """
+    else:
+        condition_concat_string = f"""
+            ml_status='Success' AND ml_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.ml_data_last_collected IS NOT NULL
+            AND ml_status!='{str(CollectionState.COLLECTING.value)}'
+            AND ml_data_last_collected <= NOW() - INTERVAL '{days_until_collect_again} DAYS'
+        """
+    
+    return condition_concat_string
+
+class CollectionRequest:
+    def __init__(self,name,phases,max_repo = 10,new_status = CollectionState.PENDING.value,additional_conditions = None,days_until_collect_again = 1):
+        self.name = name
```

Comment:
I don't think new_status should be a parameter that the constructor takes, since it is derivable from the name. And we aren't currently every passing it into the constructor we are either keeping the default parameter value or deriving it from the fact that it is a facade collection request.

I don't think additional_conditions should be a parameter that the constructor takes either because we aren't using the value

Reply:
Agreed